components/core/src/crypto/keys/mod.rs

```diff
@@ -202,13 +207,35 @@ fn check_filename(keyname: &str, filename: String, candidates: &mut HashSet<Stri
 
     if name == keyname {
         let thiskey = format!("{}-{}", name, rev);
-        candidates.insert(thiskey);
+
+        match pair_type {
+            Some(&PairType::Secret) => {
+                if suffix == SECRET_SIG_KEY_SUFFIX || suffix == SECRET_BOX_KEY_SUFFIX ||
+                    suffix == SECRET_SYM_KEY_SUFFIX
+                {
+                    candidates.insert(thiskey);
+                }
+            }
+            Some(&PairType::Public) => {
+                if suffix == PUBLIC_KEY_SUFFIX {
+                    candidates.insert(thiskey);
+                }
+            }
+            None => {
+                candidates.insert(thiskey);
+                () // this feels gross but it was the only way to make the match arms the same
```

Comment:
maybe return a bool instead, and trigger the insert on the bool value?
eg - 
let do_insert = match pair_type {
   Some(&PairType::Secret) => expression,
   Some(&PairType::Public) => expression,
   None => true
};